sdk/core/System.ClientModel/tests/client/ClientShared/ModelReaderWriterExtensions.cs

```diff
@@ -0,0 +1,264 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Text.Json;
+
+namespace ClientModel.Tests.ClientShared;
```

Comment:
Moving these files out of ModelReaderWriter folder into a "ClientShared" folder to simulate shared source/generated files used across clients (we plan to add test clients in this test project).  Tracking what to eventually do with this collection of files in https://github.com/Azure/azure-sdk-for-net/issues/40571